tests/test_settings.py

```diff
@@ -64,6 +66,27 @@ def test_save(self, env_file):
             "XANADU_CLOUD_TLS": "False",
         }
 
+    def test_save_bad_Base64URL(self, env_file):
+        """Tests that a ValueError will be raised
+        if REFRESH_TOKEN contains a character outside the Base64URL with
+        the addition that the '.' dot character is part of the token as a
+        section separator.
+        """
+        settings = xcc.Settings(
+            REFRESH_TOKEN="j.w.t\n", HOST="example.com", PORT=80, TLS=False
+        )
+        match = r"REFRESH_TOKEN contains non-Base64URL character\(s\)"
+        with pytest.raises(ValueError, match=match):
+            settings.save()
+
+        # # Check that the .env file was not modified since there was a "\n" in the refresh token.
+        # assert dotenv_values(env_file.name) == {
+        #     "XANADU_CLOUD_REFRESH_TOKEN": "j.w.t",
+        #     "XANADU_CLOUD_HOST": "example.com",
+        #     "XANADU_CLOUD_PORT": "80",
+        #     "XANADU_CLOUD_TLS": "False",
+        # }
```

Comment:
As per https://github.com/XanaduAI/xanadu-cloud-client/pull/31#issuecomment-1147937604:
```suggestion
    def test_save_bad_Base64URL(self, settings):
        """Tests that a ValueError will be raised
        if REFRESH_TOKEN contains a character outside the Base64URL with
        the addition that the '.' dot character is part of the token as a
        section separator.
        """
        settings.REFRESH_TOKEN = "j.w.t\n"

        match = r"REFRESH_TOKEN contains non-JWT character\(s\)"
        with pytest.raises(ValueError, match=match):
            settings.save()

        # Check that the .env file was not modified since there was a "\n" in the refresh token.
        assert dotenv_values(env_file.name) == {
            "XANADU_CLOUD_REFRESH_TOKEN": "j.w.t",
            "XANADU_CLOUD_HOST": "example.com",
            "XANADU_CLOUD_PORT": "80",
            "XANADU_CLOUD_TLS": "False",
        }
```